internal/middleware/gateway_check.go

```diff
@@ -0,0 +1,13 @@
+package middleware
+
+import "github.com/gofiber/fiber/v2"
+
+func NewGatewayCheck() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		headers := c.GetReqHeaders()
+		if headers["X-From-Gateway"] != "true" {
+			return c.SendStatus(fiber.StatusUnauthorized)
```

Comment:
I couldn't find a place to set the 'X-From-Gateway' header, but when will this header be set?